mbedtls/tests/async_session.rs

```diff
@@ -28,17 +31,68 @@ use support::entropy::entropy_new;
 use support::keys;
 use tokio::net::TcpStream;
 
-// TODO: Add unified interface for TCP and UDP like `TransportType` in
-// `client_server.rs`
+#[async_trait]
+trait TransportType: Sized {
+    fn get_transport_type() -> Transport;
+
+    async fn recv(ctx: &mut Context<Self>, buf: &mut [u8]) -> TlsResult<usize>;
+    async fn send(ctx: &mut Context<Self>, buf: &[u8]) -> TlsResult<usize>;
+}
+
+#[async_trait]
+impl TransportType for TcpStream {
+    fn get_transport_type() -> Transport {
+        Transport::Stream
+    }
+
+    async fn recv(ctx: &mut Context<Self>, buf: &mut [u8]) -> TlsResult<usize> {
+        ctx.read(buf).await.map_err(|_| Error::NetRecvFailed)
+    }
+
+    async fn send(ctx: &mut Context<Self>, buf: &[u8]) -> TlsResult<usize> {
+        ctx.write(buf).await.map_err(|_| Error::NetSendFailed)
+    }
+}
+
+#[async_trait]
+impl TransportType for ConnectedAsyncUdpSocket {
+    fn get_transport_type() -> Transport {
+        Transport::Datagram
+    }
+
+    async fn recv(ctx: &mut Context<Self>, buf: &mut [u8]) -> TlsResult<usize> {
```

Comment:
The `recv` and `send` here is same to `TcpStream` because I choose to also Impl `AsyncRead` and `AsyncWrite` for `ConnectedAsyncUdpSocket`